extensions-core/datasketches/src/main/java/io/druid/query/aggregation/datasketches/tuple/ArrayOfDoublesSketchBuildAggregator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.datasketches.tuple;
+
+import com.yahoo.sketches.tuple.ArrayOfDoublesUpdatableSketch;
+import com.yahoo.sketches.tuple.ArrayOfDoublesUpdatableSketchBuilder;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.segment.BaseDoubleColumnValueSelector;
+import io.druid.segment.DimensionSelector;
+import io.druid.segment.data.IndexedInts;
+
+import java.util.List;
+
+public class ArrayOfDoublesSketchBuildAggregator implements Aggregator
+{
+
+  private final DimensionSelector keySelector;
+  private final List<BaseDoubleColumnValueSelector> valueSelectors;
+
+  private ArrayOfDoublesUpdatableSketch sketch;
+
+  public ArrayOfDoublesSketchBuildAggregator(
+      final DimensionSelector keySelector, final List<BaseDoubleColumnValueSelector> valueSelectors,
+      final int nominalEntries)
+  {
+    this.keySelector = keySelector;
+    this.valueSelectors = valueSelectors;
+    sketch = new ArrayOfDoublesUpdatableSketchBuilder().setNominalEntries(nominalEntries)
+        .setNumberOfValues(valueSelectors.size()).build();
+  }
+
+  @Override
+  public synchronized void aggregate()
+  {
+    final IndexedInts keys = keySelector.getRow();
+    if (keys == null) {
+      return;
+    }
+    final double[] values = new double[valueSelectors.size()];
+    int valueIndex = 0;
+    for (final BaseDoubleColumnValueSelector valueSelector : valueSelectors) {
+      values[valueIndex++] = valueSelector.getDouble();
+    }
+    for (int i = 0; i < keys.size(); i++) {
+      final String key = keySelector.lookupName(keys.get(i));
+      sketch.update(key, values);
+    }
+  }
+
+  @Override
+  public synchronized Object get()
```

Comment:
I am not entirely sure it is necessary. I used the Theta sketch aggregator as a reference since it was developed years ago and has been extensively used in production. There a synchronized wrapper is used around the sketch. Here, instead of using a wrapper, I synchronized all methods that modify the sketch.